vector/src/main/java/im/vector/riotx/features/attachments/preview/AttachmentsPreviewFragment.kt

```diff
@@ -0,0 +1,218 @@
+            Timber.v("Crop error")
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.attachmentsPreviewRemoveAction -> {
+                handleRemoveAction()
+                true
+            }
+            R.id.attachmentsPreviewEditAction   -> {
+                handleEditAction()
+                true
+            }
+            else                                -> {
+                super.onOptionsItemSelected(item)
+            }
+        }
+    }
+
+    override fun getMenuRes() = R.menu.vector_attachments_preview
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        attachmentPreviewerMiniatureList.cleanup()
+        attachmentPreviewerBigList.cleanup()
+    }
+
+    override fun invalidate() = withState(viewModel) { state ->
+        if (state.attachments.isEmpty()) {
+            requireActivity().setResult(RESULT_CANCELED)
+            requireActivity().finish()
+        } else {
+            attachmentMiniaturePreviewController.setData(state)
+            attachmentBigPreviewController.setData(state)
+            attachmentPreviewerBigList.scrollToPosition(state.currentAttachmentIndex)
+            attachmentPreviewerMiniatureList.scrollToPosition(state.currentAttachmentIndex)
+            attachmentPreviewerSendImageOriginalSize.text = resources.getQuantityString(R.plurals.send_images_with_original_size, state.attachments.size)
+        }
+    }
+
+    override fun onAttachmentClicked(position: Int, contentAttachmentData: ContentAttachmentData) {
+        viewModel.handle(AttachmentsPreviewAction.SetCurrentAttachment(position))
+    }
+
+    private fun setResultAndFinish() = withState(viewModel) {
+        (requireActivity() as? AttachmentsPreviewActivity)?.setResultAndFinish(
+                it.attachments,
+                attachmentPreviewerSendImageOriginalSize.isChecked
+        )
+    }
+
+    private fun applyInsets() {
+        view?.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_STABLE or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+        ViewCompat.setOnApplyWindowInsetsListener(attachmentPreviewerBottomContainer) { v, insets ->
+            v.updatePadding(bottom = insets.systemWindowInsetBottom)
+            insets
+        }
+        ViewCompat.setOnApplyWindowInsetsListener(attachmentPreviewerToolbar) { v, insets ->
+            v.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+                topMargin = insets.systemWindowInsetTop
+            }
+            insets
+        }
+    }
+
+    private fun handleCropResult(result: Intent) {
+        val resultPath = UCrop.getOutput(result)?.path
+        if (resultPath != null) {
+            viewModel.handle(AttachmentsPreviewAction.UpdatePathOfCurrentAttachment(resultPath))
+        } else {
+            Toast.makeText(requireContext(), "Cannot retrieve cropped value", Toast.LENGTH_SHORT).show()
+        }
+    }
```

Comment:
Your suggestion is actually a "dangerous" practice, because if the result of the `let` block is null, the `run` block will also be executed, and it's generally not expected.

Sometimes a good old simple if/else pattern is better :)